### PR TITLE
Feature/google sheets datasource

### PR DESCRIPTION
**Pull request by szdrasiak**

### CODE REVIEW

File: frontend/schemaCMS/src/modules/dataSource/dataSource.constants.js [hunk 1]
```diff
@@ -6,12 +6,14 @@ export const FETCH_LIST_DELAY = 5 * 1000;
 export const DATA_SOURCE_FORM = 'data_source_form';
 export const DATA_SOURCE_NAME = 'name';
 export const DATA_SOURCE_FILE = 'file';
+export const DATA_SOURCE_LINK = 'link';
```
Comment (cpBurn) [inline frontend/schemaCMS/src/modules/dataSource/dataSource.constants.js:9]: if you're using this in a form, this name 'link' must be the same in the backend

File: frontend/schemaCMS/src/modules/dataSource/dataSource.constants.js [hunk 2]
```diff
@@ -6,12 +6,14 @@ export const FETCH_LIST_DELAY = 5 * 1000;
...
 export const DATA_SOURCE_TYPE = 'type';
 export const DATA_SOURCE_RUN_LAST_JOB = 'runLastJob';
 
 export const SOURCE_TYPE_FILE = 'file';
 export const SOURCE_TYPE_API = 'api';
 export const SOURCE_TYPE_DATABASE = 'database';
+export const SOURCE_TYPE_GOOGLE_SPREADSHEET = 'link';
```
Comment (cpBurn) [inline frontend/schemaCMS/src/modules/dataSource/dataSource.constants.js:16]: is it correct that we have duplicated link name?

File: frontend/schemaCMS/src/shared/components/sourceForm/sourceForm.component.js [hunk 3]
```diff
@@ -98,6 +104,22 @@ export class SourceFormComponent extends PureComponent {
     />
   );
 
+  renderGoogleUploader = ({ setFieldValue, fileName, disabled, ...restProps }) => {
```
Comment (cpBurn) [inline frontend/schemaCMS/src/shared/components/sourceForm/sourceForm.component.js:107]: I don't think this is a great name, we don't upload anything to Google. I think `renderSpreadsheetInput` explains it better

Comment (tomfelix-apptension) [reply frontend/schemaCMS/src/shared/components/sourceForm/sourceForm.component.js]: fixed

File: frontend/schemaCMS/src/modules/dataSource/dataSource.constants.js (hunk 1 above)
Comment (tomfelix-apptension) [reply frontend/schemaCMS/src/modules/dataSource/dataSource.constants.js:9]: fixed

File: frontend/schemaCMS/src/modules/dataSource/dataSource.constants.js (hunk 2 above)
Comment (tomfelix-apptension) [reply frontend/schemaCMS/src/modules/dataSource/dataSource.constants.js:16]: fixed

File: frontend/schemaCMS/src/routes/project/createDataSource/createDataSource.component.js [hunk 4]
```diff
@@ -26,6 +27,11 @@ export class CreateDataSource extends PureComponent {
     createDataSource: PropTypes.func.isRequired,
   };
 
+  validateFile = values =>
+    cond([[equals('file'), always(!values.fileName)], [equals('googleSheet'), always(!values.googleSheet)]])(
```
Comment (cpBurn) [inline frontend/schemaCMS/src/routes/project/createDataSource/createDataSource.component.js:31]: we have constants for 'file' and 'googleSheet', please use them

**Review by cpBurn — CHANGES_REQUESTED** [review]

**Comment by cpBurn** [comment]
@tomfelix-apptension can you please add tests?

File: frontend/schemaCMS/src/routes/project/createDataSource/createDataSource.component.js (hunk 4 above)
Comment (tomfelix-apptension) [reply frontend/schemaCMS/src/routes/project/createDataSource/createDataSource.component.js]: fixed

**Review by cpBurn — APPROVED** [review]

**Pull request merged**